tests/unit/test_api.py

```diff
@@ -334,117 +335,109 @@ def test_get_ctms_by_alt_id_none_found(client, dbsession, alt_id_name, alt_id_va
-    saved = get_contacts_by_any_id(dbsession, email_id=email_id)
-    assert len(saved) == 1
-    saved_contact = ContactSchema(**saved[0])
-    assert saved_contact.email == sample.email
+    saved_contacts, sample, email_id = add_contact()
+    _compare_written_contacts(saved_contacts[0], sample, email_id)
 
 
-def test_create_basic_idempotent(client, dbsession):
+@pytest.mark.parametrize("add_contact", SAMPLE_CONTACTS.keys(), indirect=True)
+def test_create_basic_idempotent(add_contact):
     """Creating a contact works across retries."""
-    email_id = UUID("d1da1c99-fe09-44db-9c68-78a75752574d")
-    sample = SAMPLE_CONTACTS[email_id]
-    resp = client.post("/ctms", sample.json())
-    assert resp.status_code == 200
-    resp = client.post("/ctms", sample.json())
-    assert resp.status_code == 200
-    saved = get_contacts_by_any_id(dbsession, email_id=email_id)
-    assert len(saved) == 1
-    saved_contact = ContactSchema(**saved[0])
-    assert saved_contact.email == sample.email
+    saved_contacts, sample, email_id = add_contact()
+    _compare_written_contacts(saved_contacts[0], sample, email_id)
+    saved_contacts, _, _ = add_contact()
+    _compare_written_contacts(saved_contacts[0], sample, email_id)
 
 
-def test_create_basic_with_id_collision(client, dbsession):
+@pytest.mark.parametrize("add_contact", SAMPLE_CONTACTS.keys(), indirect=True)
+def test_create_basic_with_id_collision(add_contact):
     """Creating a contact with the same id but different data fails."""
-    email_id = UUID("d1da1c99-fe09-44db-9c68-78a75752574d")
-    sample = SAMPLE_CONTACTS[email_id]
-    resp = client.post("/ctms", sample.json())
-    assert resp.status_code == 200
-    sample.email.mailing_country = "mx"
-    resp = client.post("/ctms", sample.json())
-    assert resp.status_code == 409
-    saved = get_contacts_by_any_id(dbsession, email_id=email_id)
-    assert len(saved) == 1
-    saved_contact = ContactSchema(**saved[0])
-    assert saved_contact.email.mailing_country == "us"
+    _, sample, _ = add_contact()
+
+    def _change_mailing(contact):
+        assert contact.email.mailing_country != "mx", "sample data has changed"
+        contact.email.mailing_country = "mx"
+        return contact
+
+    # We set check_written to False because the rows it would check for normally
+    # are actually here due to the first write
+    saved_contacts, _, _ = add_contact(
+        modifier=_change_mailing, code=409, check_redirect=False, check_written=False
+    )
+    assert saved_contacts[0].email.mailing_country == sample.email.mailing_country
 
 
-def test_create_basic_with_basket_collision(client, dbsession):
+@pytest.mark.parametrize("add_contact", SAMPLE_CONTACTS.keys(), indirect=True)
+def test_create_basic_with_basket_collision(add_contact):
     """Creating a contact with diff ids but same email fails.
     We override the basket token so that we know we're not colliding on that here.
     See other test for that check
```

Comment:
Old code, but could this comment be
```suggestion
    See other test_create_basic_with_email_collision below for that check
```